yt/data_objects/construction_data_containers.py

```diff
@@ -995,14 +997,15 @@ def _fill_sph_particles(self, fields):
 
         smoothing_style = getattr(self.ds, "sph_smoothing_style", "scatter")
         normalize = getattr(self.ds, "use_sph_normalization", True)
+        kernel_name = getattr(self.ds, "kernel_name", "cubic")
 
         bounds, size = self._get_grid_bounds_size()
 
         period = self.ds.coordinates.period.copy()
         if hasattr(period, "in_units"):
             period = period.in_units("code_length").d
-        # TODO maybe there is a better way of handling this
-        is_periodic = int(any(self.ds.periodicity))
+        # check periodicity per dimension
+        is_periodic = self.ds.periodicity
```

Comment:
Is `self.ds.periodicity` a list? I don't understand the removal of `any()` here. If it's a list, this will always be truthy.

Reply:
I changed the backend so it can accept either a single boolean or a list/array/whatever of length 3. That allows consistent handling of simulations that are periodic in some, but not all, coordinate directions. I handle those options in  [these lines](https://github.com/yt-project/yt/blob/0580c692ad36035a97fa3561cd34f891ad02b7c3/yt/utilities/lib/pixelization_routines.pyx#L1699C1-L1712C1) in `pixelize_sph_kernel_arbitrary_grid`, file: `yt/utilities/lib/pixelization_routines.pyx`. Basically, it allows backwards compatibility for single-value bool inputs in the backend, but it isn't using `if check_period` or something like that.